src/mavlink-router/endpoint.cpp

```diff
@@ -320,8 +320,8 @@ bool Endpoint::accept_msg(int target_sysid, int target_compid, uint8_t src_sysid
         }
     }
 
-    // This endpoint sent the message, and there's no other sys_comp_id: reject msg
-    if (has_sys_comp_id(src_sysid, src_compid) && _sys_comp_ids.size() == 1)
+    // This endpoint sent the message
+    if (has_sys_comp_id(src_sysid, src_compid))
```

Comment:
As you mentioned in description it will drop packet though there are other component IDs connected to same endpoint. @edersondisouza mentioned in #113 that he observed a scenario where different components are connected to same end point while working on ROS.

Reply:
It's ok for different components to be connected to the same endpoint. This is normal. What's not normal is that these components depend on mavlink-router to work.

Examples:

```
GCS -> Flight Controller   -> mavlink-router -> App
       Flight Controller -/
```

```
GCS -> Flight Controller   -> mavlink-router -> mavros -> App1
       Flight Controller -/                            \-> App2
```

In the first case, the flight stack on the flight controller needs to check if it should handle the message or if it should forward it. Both ArduPilot and PX4 already do that

On the second case, App1 and App2 should not be talking to each other via mavlink-router. They should either a) be communicating directing (alas they are ROS applications and should keep the mavlink serialization and de-serialization out of the game) or b) make mavros route the packet. There's no way for mavlink-router to do the right thing if it sends a (possibly broadcast message) over to the same endpoint that sent it.

Reply:
I think it's fine, then.